### ISSUE TITLE
All Web Smoke tests fail for me under Linux

### ISSUE THREAD

**Issue by dbaeumer** [microsoft/vscode#127409]
Testing #127395

The Web smoke test all failed for me. @aeschli tried as well on his native Linux machine with the same outcome.

Insider build used: `cc52823098`

~~According to @aeschli smoke test passes with latest stable (1.57.1)~~

Assigning to @bpasero and @lszomoru

**Comment by bpasero** [comment]
Care to share what happens?

Btw web smoke tests run in our CI also on Linux: https://monacotools.visualstudio.com/DefaultCollection/Monaco/_build/results?buildId=125485&view=logs&j=4d2898ab-dfbe-557e-92e7-aaac158fdd2f&t=500500fc-d3cc-5138-3a2a-6935e05fea33

**Comment by dbaeumer** [comment]
Here is what I see

```
dirkb:~/Projects/../vscode [main|…1] $ yarn smoketest --build ~/bin/vscode-server-linux-x64-web --web --browser chromium
yarn run v1.22.5
$ cd test/smoke && yarn compile && node test/index.js --build /home/dirkb/bin/vscode-server-linux-x64-web --web --browser chromium
$ yarn --cwd ../automation compile && tsc
$ npm run copy-driver && npm run copy-driver-definition && tsc

> vscode-automation@1.54.0 copy-driver
> cpx src/driver.js out/


> vscode-automation@1.54.0 copy-driver-definition
> node tools/copy-driver-definition.js

Running web smoke tests against /home/dirkb/bin/vscode-server-linux-x64-web


*** Test data: /tmp/t1484jESFdWcR26oT
*** Preparing smoketest setup...
*** Cloning test project repository...
*** Smoketest setup done!

  VSCode Smoke Tests (Web)
    VSCode Smoke Tests (Web)
      Search
Starting built server from '/home/dirkb/bin/vscode-server-linux-x64-web/server.sh'
Storing log files into '/home/dirkb/Projects/mseng/VSCode/vscode/.build/logs/smoke-tests-browser'
        - has a tooltp with a keybinding
** Timeout!
Error: Element not found for textContent
    at /home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:238:159
    at async poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:181:22)
    at async Code.waitForTextContent (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:238:16)
    at async Search.waitForResultText (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:91:9)
    at async Context.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/test/smoke/out/areas/search/search.test.js:32:13)
        1) searches for body & checks for correct result number
** Timeout!
Error: Element not found for textContent
    at /home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:238:159
    at async poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:181:22)
    at async Code.waitForTextContent (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:238:16)
    at async Search.waitForResultText (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:91:9)
    at async Context.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/test/smoke/out/areas/search/search.test.js:40:13)
        2) searches only for *.js files & checks for correct result number
        - dismisses result & checks for correct result number
** Timeout!
page.evaluate: Evaluation failed: Error: Element not found: .search-view .results .filematch[data-resource$="app.js"]
    at M._getElementXY (http://localhost:8000/static/out/vs/workbench/workbench.web.api.js:627:19754)
    at M.getElementXY (http://localhost:8000/static/out/vs/workbench/workbench.web.api.js:627:18750)
    at eval (eval at evaluate (:303:29), <anonymous>:1:15)
    at eval (<anonymous>)
    at UtilityScript.evaluate (<anonymous>:303:29)
    at UtilityScript.<anonymous> (<anonymous>:1:44)
Error
    at Object.captureStackTrace (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/utils/stackTrace.js:50:19)
    at Connection.sendMessageToServer (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/connection.js:65:48)
    at Proxy.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/channelOwner.js:64:61)
    at /home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/frame.js:184:42
    at Frame._wrapApiCall (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/channelOwner.js:77:34)
    at Frame.evaluate (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/frame.js:183:21)
    at /home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/page.js:395:60
    at Page._attributeToPage (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/page.js:238:20)
    at Page.evaluate (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/page.js:395:21)
    at Object.getElementXY (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/playwrightDriver.js:78:70)
    at Proxy.click (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/playwrightDriver.js:65:43)
    at Proxy.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:212:39)
    at /home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:242:38
    at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:181:28)
    at async Code.waitAndClick (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:242:9)
    at async retry (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:16:9)
    at async Search.replaceFileMatch (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:84:9)
    at async Context.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/test/smoke/out/areas/search/search.test.js:55:13)
        3) replaces first search result with a replace term
      Quick Access
Starting built server from '/home/dirkb/bin/vscode-server-linux-x64-web/server.sh'
Storing log files into '/home/dirkb/Projects/mseng/VSCode/vscode/.build/logs/smoke-tests-browser'
** Timeout!
Did not pass accept function
        4) quick access search produces correct result
** Timeout!
Did not pass accept function
        5) quick access respects fuzzy matching
      Notebooks
        - inserts/edits code cell
        - inserts/edits markdown cell
        - moves focus as it inserts/deletes a cell
        - moves focus in and out of output
        - cell action execution
      Language Features
Starting built server from '/home/dirkb/bin/vscode-server-linux-x64-web/server.sh'
Storing log files into '/home/dirkb/Projects/mseng/VSCode/vscode/.build/logs/smoke-tests-browser'
** Timeout!
Did not pass accept function
        6) verifies quick outline
** Timeout!
Did not pass accept function
        7) verifies problems view
** Timeout!
Did not pass accept function
        8) verifies settings
      Editor
Starting built server from '/home/dirkb/bin/vscode-server-linux-x64-web/server.sh'
Storing log files into '/home/dirkb/Projects/mseng/VSCode/vscode/.build/logs/smoke-tests-browser'
** Timeout!
Did not pass accept function
        9) shows correct quick outline
      Statusbar
Starting built server from '/home/dirkb/bin/vscode-server-linux-x64-web/server.sh'
Storing log files into '/home/dirkb/Projects/mseng/VSCode/vscode/.build/logs/smoke-tests-browser'
** Timeout!
Did not pass accept function
        10) verifies presence of all default status bar elements
** Timeout!
page.evaluate: Evaluation failed: Error: Element not found: .statusbar-item[id="status.scm"] .codicon.codicon-git-branch
    at M._getElementXY (http://localhost:8000/static/out/vs/workbench/workbench.web.api.js:627:19754)
    at M.getElementXY (http://localhost:8000/static/out/vs/workbench/workbench.web.api.js:627:18750)
    at eval (eval at evaluate (:303:29), <anonymous>:1:15)
    at eval (<anonymous>)
    at UtilityScript.evaluate (<anonymous>:303:29)
    at UtilityScript.<anonymous> (<anonymous>:1:44)
Error
    at Object.captureStackTrace (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/utils/stackTrace.js:50:19)
    at Connection.sendMessageToServer (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/connection.js:65:48)
    at Proxy.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/channelOwner.js:64:61)
    at /home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/frame.js:184:42
    at Frame._wrapApiCall (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/channelOwner.js:77:34)
    at Frame.evaluate (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/frame.js:183:21)
    at /home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/page.js:395:60
    at Page._attributeToPage (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/page.js:238:20)
    at Page.evaluate (/home/dirkb/Projects/mseng/VSCode/vscode/node_modules/playwright/lib/client/page.js:395:21)
    at Object.getElementXY (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/playwrightDriver.js:78:70)
    at Proxy.click (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/playwrightDriver.js:65:43)
    at Proxy.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:212:39)
    at /home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:242:38
    at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:181:28)
    at async Code.waitAndClick (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:242:9)
    at async StatusBar.clickOn (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/statusbar.js:17:9)
    at async Context.<anonymous> (/home/dirkb/Projects/mseng/VSCode/vscode/test/smoke/out/areas/statusbar/statusbar.test.js:33:13)
        11) verifies that 'quick input' opens when clicking on status bar elements
        ✓ verifies that 'Problems View' appears when clicking on 'Problems' status element
        ✓ verifies that 'Tweet us feedback' pop-up appears when clicking on 'Feedback' icon
** Timeout!
Did not pass accept function
        12) checks if 'Go to Line' works if called from the status bar
** Timeout!
Did not pass accept function
        13) verifies if changing EOL is reflected in the status bar
      Extensions
Starting built server from '/home/dirkb/bin/vscode-server-linux-x64-web/server.sh'
Storing log files into '/home/dirkb/Projects/mseng/VSCode/vscode/.build/logs/smoke-tests-browser'
** Timeout!
Did not pass accept function
        14) install and enable vscode-smoketest-check extension


  2 passing (5m)
  7 pending
  14 failing

  1) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Search
           searches for body & checks for correct result number:
     Error: Timeout: get text content '.search-view .messages .message' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForTextContent (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:238:16)
      at async Search.waitForResultText (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:91:9)
      at async Context.<anonymous> (out/areas/search/search.test.js:32:13)

  2) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Search
           searches only for *.js files & checks for correct result number:
     Error: Timeout: get text content '.search-view .messages .message' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForTextContent (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:238:16)
      at async Search.waitForResultText (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:91:9)
      at async Context.<anonymous> (out/areas/search/search.test.js:40:13)

  3) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Search
           replaces first search result with a replace term:
     Error: Timeout: click '.search-view .results .filematch[data-resource$="app.js"]' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitAndClick (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:242:9)
      at async retry (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:16:9)
      at async Search.replaceFileMatch (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/search.js:84:9)
      at async Context.<anonymous> (out/areas/search/search.test.js:55:13)

  4) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Quick Access
           quick access search produces correct result:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async Context.<anonymous> (out/areas/search/search.test.js:78:13)

  5) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Quick Access
           quick access respects fuzzy matching:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async Context.<anonymous> (out/areas/search/search.test.js:89:13)

  6) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Language Features
           verifies quick outline:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async QuickAccess.openFile (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickaccess.js:42:9)
      at async Context.<anonymous> (out/areas/languages/languages.test.js:16:13)

  7) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Language Features
           verifies problems view:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async QuickAccess.openFile (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickaccess.js:42:9)
      at async Context.<anonymous> (out/areas/languages/languages.test.js:22:13)

  8) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Language Features
           verifies settings:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async QuickAccess.openFile (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickaccess.js:42:9)
      at async Context.<anonymous> (out/areas/languages/languages.test.js:32:13)

  9) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Editor
           shows correct quick outline:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async QuickAccess.openFile (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickaccess.js:42:9)
      at async Context.<anonymous> (out/areas/editor/editor.test.js:15:13)

  10) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Statusbar
           verifies presence of all default status bar elements:
     Error: Timeout: get element '.statusbar-item[id="status.scm"] .codicon.codicon-git-branch' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElement (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:258:16)
      at async StatusBar.waitForStatusbarElement (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/statusbar.js:14:9)
      at async Context.<anonymous> (out/areas/statusbar/statusbar.test.js:15:13)

  11) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Statusbar
           verifies that 'quick input' opens when clicking on status bar elements:
     Error: Timeout: click '.statusbar-item[id="status.scm"] .codicon.codicon-git-branch' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitAndClick (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:242:9)
      at async StatusBar.clickOn (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/statusbar.js:17:9)
      at async Context.<anonymous> (out/areas/statusbar/statusbar.test.js:33:13)

  12) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Statusbar
           checks if 'Go to Line' works if called from the status bar:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async QuickAccess.openFile (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickaccess.js:42:9)
      at async Context.<anonymous> (out/areas/statusbar/statusbar.test.js:68:13)

  13) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Statusbar
           verifies if changing EOL is reflected in the status bar:
     Error: Timeout: get elements '.quick-input-widget .quick-input-list .monaco-list-row .label-name' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:254:16)
      at async QuickInput.waitForQuickInputElements (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickinput.js:25:9)
      at async QuickAccess.openFile (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/quickaccess.js:42:9)
      at async Context.<anonymous> (out/areas/statusbar/statusbar.test.js:76:13)

  14) VSCode Smoke Tests (Web)
       VSCode Smoke Tests (Web)
         Extensions
           install and enable vscode-smoketest-check extension:
     Error: Timeout: get element 'div.extensions-viewlet[id="workbench.view.extensions"] .monaco-list-row[data-extension-id="michelkaporin.vscode-smoketest-check"]' after 20 seconds.
      at poll (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:177:19)
      at async Code.waitForElement (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/code.js:258:16)
      at async Extensions.searchForExtension (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/extensions.js:27:9)
      at async Extensions.installExtension (/home/dirkb/Projects/mseng/VSCode/vscode/test/automation/out/extensions.js:37:9)
      at async Context.<anonymous> (out/areas/extensions/extensions.test.js:19:13)



error Command failed with exit code 255.
```

**Comment by dbaeumer** [comment]
And here the log file

[remoteagent.log](https://github.com/microsoft/vscode/files/6732210/remoteagent.log)

**Comment by aeschli** [comment]
For me the web tests went through on Linux

**Comment by bpasero** [comment]
Same for me. @dbaeumer did you possibly not `tsc` in `test/automation` and `test/smoke` before running? I think we do not compile those folders automatically.

**Comment by dbaeumer** [comment]
@bpasero I did. I even ran the normal smoke test successfully.

**Comment by dbaeumer** [comment]
I re-ran everything (compile, even compile in test/automation and test/smoke by hand) and the result is still the same. I noticed that the test get stuck in

<img width="783" alt="capture" src="https://user-images.githubusercontent.com/1931590/123805828-b4ad3b80-d8ee-11eb-97a3-541efb7de1ac.png">

**Comment by dbaeumer** [comment]
This is all under WSL in case it matters

**Comment by dbaeumer** [comment]
I ran it again to check if the search path exist and it does
<img width="801" alt="capture" src="https://user-images.githubusercontent.com/1931590/123806494-4b79f800-d8ef-11eb-8f62-8d063835291a.png">

**Comment by bpasero** [comment]
@dbaeumer wait, you run the smoke test from WSL? not from a bare Linux? that sounds like important info, I doubt we ever supported that?

@aeschli @joaomoreno did you ever run web smoke tests in WSL and did it work?

**Comment by joaomoreno** [comment]
I've been doing that for well over a year now. As long as there's a functional X server, they run just fine. Both besktop and web.

**Comment by bpasero** [comment]
Ok thanks, then maybe Dirk and you could chat how to unblock him. It looks like this maybe a setup problem.

**Comment by joaomoreno** [comment]
@dbaeumer Let's jump into a call?

**Comment by dbaeumer** [comment]
@bpasero as I outlines earlier I can run all the other smoke tests. And the error I posted here https://github.com/microsoft/vscode/issues/127409#issuecomment-870603991 looks wired to me and has nothing todo with X-Server setup

**Comment by joaomoreno** [comment]
This appears related to @lszomoru's good change to isolate each suite in its own session. We should find where that breaks down.

**Comment by dbaeumer** [comment]
With the help of @joaomoreno I was able to capture more lock information:

```
Server stdout: [11:36:37] [::1][c1902ed4][ManagementConnection] New connection established.

** getElements .monaco-workbench
Server stdout: [11:36:38] [::1][f5460508][ExtensionHostConnection] New connection established.

Server stdout: [11:36:38] [::1][f5460508][ExtensionHostConnection] <9553> Launched Extension Host Process.

Server stdout: [11:36:38] [::1][f5460508][ExtensionHostConnection] <9553> Extension Host Process exited with code: 0, signal: null.

Server stderr: [11:36:38] [::1][f5460508][ExtensionHostConnection] Unknown reconnection token.

Server stdout: [11:36:39] [::1][c1902ed4][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.
```

```
 INFO [remote-connection][ExtensionHost][0ccf4…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
log.ts:296   ERR CloseEvent
log.ts:284  INFO [remote-connection][ExtensionHost][0ccf4…][reconnect] starting reconnecting loop. You can get more information with the trace log level.
log.ts:284  INFO [remote-connection][ExtensionHost][0ccf4…][reconnect] resolving connection...
log.ts:284  INFO [remote-connection][ExtensionHost][0ccf4…][reconnect] connecting to localhost:8000...
log.ts:284  INFO [remote-connection][ExtensionHost][0ccf4…][reconnect] received socket close event (wasClean: false, code: 1006, reason: ).
log.ts:296   ERR CloseEvent
log.ts:296   ERR [remote-connection][ExtensionHost][0ccf4…][reconnect][localhost:8000] received error control message when negotiating connection. Error:
log.ts:296   ERR Error: Connection error: Unknown reconnection token
    at W (remoteAgentConnection.ts:717)
    at remoteAgentConnection.ts:313
    at a.fire (event.ts:583)
    at o.fire (ipc.net.ts:513)
    at h._receiveMessage (ipc.net.ts:870)
    at ipc.net.ts:815
    at a.fire (event.ts:583)
    at v.acceptChunk (ipc.net.ts:286)
    at ipc.net.ts:247
    at browserSocketFactory.ts:197
```

So basically the extension host shuts down immediately.

**Comment by dbaeumer** [comment]
We made sure that all native modules can be loaded in the node version we ship with the server

```
Welcome to Node.js v14.16.0.
Type ".help" for more information.
> require('spdlog')
{
  version: 10805,
  setLevel: [Function: setLevel],
  setFlushOn: [Function: setFlushOn],
  Logger: [Function: Logger],
  createRotatingLogger: [Function: createRotatingLogger],
  createAsyncRotatingLogger: [Function: createAsyncRotatingLogger]
}
> require('vsda')
{ signer: [Function: signer], validator: [Function: validator] }
> require('node-pty')
{
  spawn: [Function: spawn],
  fork: [Function: fork],
  createTerminal: [Function: createTerminal],
  open: [Function: open],
  native: {
    fork: [Function (anonymous)],
    open: [Function (anonymous)],
    resize: [Function (anonymous)],
    process: [Function (anonymous)]
  }
}
> require('native-watchdog')
{ start: [Function (anonymous)], exit: [Function (anonymous)] }
> require('nfsw')
Uncaught Error: Cannot find module 'nfsw'
Require stack:
- <repl>
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '<repl>' ]
}
> require('nsfw')
[Function: nsfw] {
  actions: { CREATED: 0, DELETED: 1, MODIFIED: 2, RENAMED: 3 },
  _native: [Function: NSFW]
}
```

@alexdima any idea why the extension host is closing immediately

**Comment by dbaeumer** [comment]
I restarted WSL and ran the test again. Same outcome.

**Comment by joaomoreno** [comment]
> This appears related to @lszomoru's good change to isolate each suite in its own session. We should find where that breaks down.

I take it back, this has nothing to do with @lszomoru's changes!

**Comment by alexdima** [comment]
This was a bit expensive (as it took me quite a while to get the setup), but here is what I observed:
- I downladed the build `cc52823098` -> `vscode-server-linux-x64-web.tar.gz`
- (I did not have WSL configured with x windows)
  - when I executed `yarn smoketest --build ~/src/vscode-server-linux-x64-web --web --browser chromium` I saw the same hangs that @dbaeumer reported
  - I tried to launch `yarn smoketest --build ~/src/vscode-server-linux-x64-web --web --browser chromium --debug` but everything kept on hanging and I could not see the browser
- so I went and followed all the steps at https://github.com/microsoft/vscode/wiki/Selfhosting-on-Windows-WSL . I did not do those steps previously
- those steps were incomplete, I also had to execute `sudo apt-get install -y libgbm-dev` to get `./scripts/code.sh` to work and I had to manually allow the xserver even on public networks 🤷‍♂️ .
- finally after doing all that setup, now the smoketest passes for me and I get to always see it even when invoking without `--debug`, so I think there might be something off with playwright under WSL where it always decides to show the browser. AFAIK it should only show the browser when invoking with `--debug`.

In any case, @dbaeumer , this works for me. Here is a video to show that it is working for me:

https://user-images.githubusercontent.com/5047891/123956421-f996a800-d9aa-11eb-9866-af2d755f7b7f.mp4

@joaomoreno I think you already have this setup, can you please confirm if you reproduce what @dbaeumer is seeing or if smoketests run for you?

**Comment by joaomoreno** [comment]
Here are my `gdm`-related installed libs:

```
➜  ~ sudo apt list --installed | grep gdm

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

gdm3/focal-updates,now 3.36.3-0ubuntu0.20.04.3 amd64 [installed,automatic]
gir1.2-gdm-1.0/focal-updates,now 3.36.3-0ubuntu0.20.04.3 amd64 [installed,automatic]
libgdm1/focal-updates,now 3.36.3-0ubuntu0.20.04.3 amd64 [installed,automatic]
```

The smoketests do run for me without any issue.

**Comment by alexdima** [comment]
> Do you think `libgbm` was the culprit? How come the server doesn't spit out the right error?

I have to be honest and admit that I have no idea :). I basically followed the steps from the WSL selfhosting guide and then launched `./scripts/code.sh` just to see how it looks. That's when I got:
```
error while loading shared libraries: libgbm.so.1: cannot open shared object file: No such file or directory
```
So I quickly made my way to https://stackoverflow.com/a/67407105 and installed `libgbm-dev`.

I cannot say if that library helps playwright or our nodejs server. My guess would be it helps playwright.

**Comment by joaomoreno** [comment]
From @dbaeumer's log, it's our server who decides to die suddenly. @dbaeumer, does it help if you install this lib?

**Comment by dbaeumer** [comment]
Not yet. Will do today.

**Comment by dbaeumer** [comment]
OK. I checked and I didn't have `libgdm-dev` installed. Would have been cool to have something in the log that points that out.

**Comment by dbaeumer** [comment]
I installed `libgdm-dev` and now have

```
dirkb:~/Projects/../vscode [main|⚑ 1] $ sudo apt list --installed | grep gdm

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

gir1.2-gdm-1.0/focal-updates,now 3.36.3-0ubuntu0.20.04.3 amd64 [installed,automatic]
libgdm-dev/focal-updates,now 3.36.3-0ubuntu0.20.04.3 amd64 [installed]
libgdm1/focal-updates,now 3.36.3-0ubuntu0.20.04.3 amd64 [installed,automatic]
```

But running the smoke test still results in the same exit of the extension host on my machine. @alexdima any hint how to find out why the extension host exists.

```
*
* Visual Studio Code Server
*
* Reminder: You may only use this software with Visual Studio family products,
* as described in the license https://aka.ms/vscode-remote/license
*

Extension host agent listening on 9000


Server stderr: Ignoring option browser: not supported for server.

Server stdout: Web UI available at http://localhost:9000/?tkn=00eaac51-e627-4a9e-b7eb-77d16088adb5

Server stdout: [10:27:03] Extension host agent started.

** getWindowIds
** getWindowIds
** getElements .monaco-workbench
Server stdout: [10:27:05] [::1][d61fb06e][ManagementConnection] New connection established.

** getElements .monaco-workbench
Server stdout: [10:27:06] [::1][ab4beed9][ExtensionHostConnection] New connection established.

Server stdout: [10:27:06] [::1][ab4beed9][ExtensionHostConnection] <3914> Launched Extension Host Process.

Server stdout: [10:27:06] [::1][ab4beed9][ExtensionHostConnection] <3914> Extension Host Process exited with code: 0, signal: null.

Server stderr: [10:27:06] [::1][ab4beed9][ExtensionHostConnection] Unknown reconnection token.

Server stdout: [10:27:06] [::1][d61fb06e][ManagementConnection] The client has disconnected gracefully, so the connection will be disposed.

```

**Comment by dbaeumer** [comment]
I reopen. Would still like to understand why it fails for me.
